src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'green' | 'red' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.isValidTriangle()) {
+      throw new Error('Impossible to make a triangle with such sides');
+    }
```

Comment:
```suggestion
    if (this.isInvalidTriangle()) {
      throw new Error('Impossible to make a triangle with such sides');
    }
```
I would rather rename the method name. Because here it sounds like "If this is valid triangle, I will throw Error". Which is kinda confusing in terms of human language. The same in other methods.

Reply:
I agree with Oleh, let's fix it.